scripts/create-secrets/prompts/inquirer-fuzzy-path.js

```diff
@@ -0,0 +1,100 @@
+const fs = require("fs");
+const path = require("path");
+const util = require("util");
+
+const Choices = require("inquirer/lib/objects/choices");
+const InquirerAutocomplete = require("inquirer-autocomplete-prompt");
+const stripAnsi = require("strip-ansi");
+const style = require("ansi-styles");
+const fuzzy = require("fuzzy");
+
+const readdir_ = util.promisify(fs.readdir);
+
+class InquirerFuzzyPath extends InquirerAutocomplete {
+    constructor(question, rl, answers) {
+        const rootPath = question.rootPath || ".";
+        const pathFilter = question.pathFilter || (() => true);
+        const _question = Object.assign({}, question, {
+            source: (_, pattern) => getPaths(rootPath, pattern, pathFilter)
+        });
+        super(_question, rl, answers);
+    }
+    search(searchTerm) {
+        return super.search(searchTerm).then(value => {
+            this.currentChoices.getChoice = choiceIndex => {
+                let choice = Choices.prototype.getChoice.call(
+                    this.currentChoices,
+                    choiceIndex
+                );
+                return {
+                    value: stripAnsi(choice.value),
+                    name: stripAnsi(choice.name),
+                    short: stripAnsi(choice.name)
+                };
+            };
+        });
+    }
+    onSubmit(line) {
+        if (typeof this.opt.validate === "function") {
+            var validationResult = this.opt.validate(
+                this.currentChoices.getChoice(this.selected)
+            );
+            if (validationResult !== true) {
+                this.render(
+                    validationResult || "Enter something, tab to autocomplete!"
+                );
+                return;
+            }
+        }
+        super.onSubmit(stripAnsi(line));
+    }
+}
+
+const maxScanDepth = 2;
```

Comment:
If this is a const can it go at the top of the file?

Reply:
👍